wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -20,32 +21,46 @@ private[wallet] trait TransactionProcessing extends WalletLogger {
   /////////////////////
   // Public facing API
 
+  def getNumberOfConfirmations(
+      blockHashOpt: Option[DoubleSha256DigestBE]): Future[Int] =
+    blockHashOpt match {
+      case None => Future.successful(0)
+      case Some(blockHash) =>
+        for {
+          blockHeight <- chainQueryApi.getBlockHeight(blockHash)
+          tipHash <- chainQueryApi.getBestBlockHash()
```

Comment:
maybe add a helper method in `ChainQueryApi` called `getBestBlockHashHeight()` ?